pkg/kwokctl/cmd/export/logs/logs.go

```diff
@@ -0,0 +1,158 @@
+	"sigs.k8s.io/kwok/pkg/utils/version"
+)
+
+type flagpole struct {
+	Name string
+}
+
+// NewCommand returns a new cobra.Command for getting the cluster logs
+func NewCommand(ctx context.Context) *cobra.Command {
+	flags := &flagpole{}
+	cmd := &cobra.Command{
+		Args: cobra.MaximumNArgs(1),
+		// TODO(bentheelder): more detailed usage
+		Use:   "logs [output-dir]",
+		Short: "Exports logs to a tempdir or [output-dir] if specified",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			flags.Name = config.DefaultCluster
+			return runE(ctx, flags, args)
+		},
+	}
+	return cmd
+}
+
+func runE(ctx context.Context, flags *flagpole, args []string) error {
+	name := config.ClusterName(flags.Name)
+	workdir := path.Join(config.ClustersDir, flags.Name)
+
+	logger := log.FromContext(ctx).With("cluster", flags.Name)
+	ctx = log.NewContext(ctx, logger)
+
+	rt, err := runtime.DefaultRegistry.Load(ctx, name, workdir)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			logger.Warn("Cluster is not exists")
+		}
+		return err
+	}
+
+	// get the optional directory argument, or create a tempdir
+	var dir string
+	if len(args) == 0 {
+		t, err := file.TempDir("", "")
+		if err != nil {
+			return err
+		}
+		dir = t
+	} else {
+		dir = args[0]
+	}
+
+	logger.Info("Exporting logs to ", dir)
+
+	// ensure directory
+	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
+		return fmt.Errorf("failed to create logs directory")
+	}
+
+	if err := os.WriteFile(
+		filepath.Join(dir, "kwok-version.txt"),
+		[]byte(version.DisplayVersion()),
+		0640,
+	); err != nil {
+		return fmt.Errorf("failed to write kwok-version.txt %w", err)
+	}
+
+	cfg, err := rt.Config(ctx)
+	if err != nil {
+		return err
+	}
+
+	err = config.Save(ctx, filepath.Join(dir, "kwok.yaml"), []config.InternalObject{cfg})
+	if err != nil {
+		return err
+	}
+
+	logToPathFn := func(path, name string) func() error {
+		return func() error {
+			f, err := file.OnHost(path)
+			if err != nil {
+				return err
+			}
+			defer func() {
+				_ = f.Close()
+				if err != nil {
+					_ = os.Remove(path)
+				}
+			}()
+			return rt.Logs(ctx, name, f)
+		}
+	}
+
+	// construct a slice of methods to collect logs
+	fns := make([]func() error, 0, len(cfg.Components)+1)
+
+	for _, component := range cfg.Components {
+		fns = append(fns, logToPathFn(filepath.Join(filepath.Join(dir, name), component.Name+".log"), component.Name))
+	}
+
+	if cfg.Options.KubeAuditPolicy != "" {
+		fns = append(
+			fns,
+			func() error {
+				filePath := filepath.Join(dir, name, "audit.log")
+				f, err := file.OnHost(filePath)
+				if err != nil {
+					return err
+				}
+				defer func() {
+					_ = f.Close()
+					if err != nil {
+						_ = os.Remove(filePath)
+					}
+				}()
+				return rt.AuditLogs(ctx, f)
+			})
+	}
+
+	fns = append(fns, rt.CollectLogs(ctx, name, dir)...)
+
+	var errs []error
+	// run and collect up all errors
+	errs = append(errs, concurrent.AggregateConcurrent(fns))
```

Comment:
IMO, sequential execution gives better performance in cases where a lot of data is being written.